rak811_test.go

```diff
@@ -407,27 +369,46 @@ func TestLora_GetRadioStatus(t *testing.T) {
 	t.Run("get the radio statistics", func(t *testing.T) {
 		res, err := lora.GetRadioStatus()
 		if err != nil {
-			t.Errorf("error %v", err)
+			t.Fatalf("error %v", err)
 		}
-		if bytes.Compare([]byte(res), fsp.buf) != 0 {
-			t.Errorf("got %q, want %q", res, []byte(fsp.buf))
+		if bytes.Compare([]byte(res), fsp.At()) != 0 {
+			t.Errorf("got %q, want %q", res, fsp.At())
 		}
 	})
 }
 
+func newFakeFakeSerialPort(buf []byte) *FakeSerialPort {
+	reader := bufio.NewReader(bytes.NewReader(buf))
+
+	return &FakeSerialPort{
+		reader: reader,
+	}
+}
+
 type FakeSerialPort struct {
-	buf []byte
+	reader *bufio.Reader
+	last   []byte
 }
 
-func (f FakeSerialPort) Read(p []byte) (n int, err error) {
-	n = copy(p, f.buf)
+func (f *FakeSerialPort) Read(p []byte) (n int, err error) {
+	line, err := f.reader.ReadBytes('\n')
+	if err != nil {
+		return 0, err
+	}
+	n = copy(p, line)
+	f.last = line
 	return n, nil
 }
 
-func (FakeSerialPort) Write(p []byte) (n int, err error) {
+func (*FakeSerialPort) Write(p []byte) (n int, err error) {
```

Comment:
This should be used as the guideline and is not that important to me in this situation, but for future reference, we should consider this list instead of just because all some receivers use pointers. 

https://github.com/golang/go/wiki/CodeReviewComments#receiver-type